src/modules/Sticky/Sticky.js

```diff
@@ -0,0 +1,230 @@
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import {
+  customPropTypes,
+  META,
+  getElementType,
+  getUnhandledProps,
+} from '../../lib'
+
+/**
+ * Sticky content stays fixed to the browser viewport while
+ * another column of content is visible on the page.
+ */
+export default class Sticky extends Component {
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Context which sticky element should stick to. */
+    context: PropTypes.object,
+
+    /** Offset in pixels from the top of the screen when fixing element to viewport. */
+    offset: PropTypes.number,
+
+    /** Offset in pixels from the bottom of the screen when fixing element to viewport. */
+    bottomOffset: PropTypes.number,
+
+    /** Whether element should be "pushed" by the viewport, attaching to the bottom of the screen when scrolling up. */
+    pushing: PropTypes.bool,
+
+    /* Callback when element is bound to bottom of parent container. */
+    onBottom: PropTypes.func,
+
+    /**
+     * Callback when element is fixed to page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onStick: PropTypes.func,
+
+    /* Callback when element is bound to top of parent container. */
+    onTop: PropTypes.func,
+
+    /**
+     * Callback when element is unfixed from page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onUnstick: PropTypes.func,
+  }
+
+  static defaultProps = {
+    offset: 0,
+    bottomOffset: 0,
+  }
+
+  static _meta = {
+    name: 'Sticky',
+    type: META.TYPES.MODULE,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = { sticky: false, top: null, bottom: null }
+  }
+
+  componentDidMount() {
+    this.update()
+    window.addEventListener('scroll', this.update)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.update)
+  }
+
+  calcBoundingRects() {
+    this.triggerBoundingRect = this.triggerRef.getBoundingClientRect()
+    this.contextBoundingRect = (this.props.context || document.body).getBoundingClientRect()
+    this.stickyBoundingRect = this.stickyRef.getBoundingClientRect()
+  }
+
+  // Return true when the component reached the bottom of the context
+  hasReachedContextBottom() {
```

Comment:
I don't think that `has` has something meaningful in the method's name, I think `reachedContextBottom` is enough clear.